### ISSUE TITLE
User Current location

### ISSUE THREAD

**Issue by ammar-sheikh** [aurovrata/cf7-google-map#5]
Hi, 
I really like this plugin but I'm wondering if I can get user current location and show a marker for this location? 

Looking forward hearing from you 
Kind regards, 
Ammar

**Comment by aurovrata** [comment]
Someone already requested this on the[ WordPress support forum](https://wordpress.org/support/topic/geolocation-22/).

This isn't something I am willing to develop currently as it would only make sense on mobile browsers, most desktop/laptop browsers are unable to geolocate accurately.

However, you can develop your own solution with this plugin as the gmap3 object is exposed.  

1.  Use the initialisation event to get a reference of the gmap3 object,
2.  Clear the existing marker, 
```
marker.setMap(null);
marker = null;
```
3.  create a [new marker at the location](https://gmap3.net/api-marker.html) required.

**Comment by aurovrata** [comment]
You can test your browser location [here](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API#Result).  It does not work for my browser due to privacy settings, and this is likely across the majority of browsers.

Developing this functionality in the plugin would not be very difficult, but it would likely end up having a lot of non-coder users complaining the plugin does not work and make support a headache.

Hence, I prefer to leave this out and give access to underlying objects for coders to implement this knowing the limitations of the functionality.

**Comment by kartguru** [comment]
What about using an IP lookup to identify rough location? User could then find the appropriate address to enter much quicker.

Cheers and nice suite of cf7 plugins

**Comment by aurovrata** [comment]
Indeed, [HTML5 Geolocation API](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API) already does (try it [here](https://bestvpn.org/html5demos/geo/)).  However, given that it uses a user's private information, it would also require special handling in the backend to comply with the EU's [GDPR](https://www.codeinwp.com/blog/complete-wordpress-gdpr-guide/) regulation.  

Given the ease to implement this on a custom basis using the plugin's exposed js initialisation process, and the low number of request for this, I prefer to leave it out of the plugin.

> it would likely end up having a lot of non-coder users complaining the plugin does not work and make support a headache.

another reason to leave it out...for now at least

**Comment by kartguru** [comment]
fair call, thanks for responding @aurovrata 

all the best

**Comment by aurovrata** [comment]
I may come back on this decision and implement in a future version....I am rather busy with the Post My CF7 Form plugin rewrite at this point of time.

Anyhow, I keep this issue open for now.
